op-service/client/dial_test.go

```diff
@@ -0,0 +1,17 @@
+package client
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestIsURLAvailable(t *testing.T) {
+	go func() {
+		_ = http.ListenAndServe(":8989", nil)
+	}()
```

Comment:
Shouldn't something stop this server again at the end of the test?

And this test is not going to be flaky when it starts hitting port conflicts. You probably need to listen on port 0 which will guarantee selecting a free port and you can then get the effective port to connect to.  Checking if a URL isn't available I'd probably use an unresolvable domain - sadly `example.com` actually resolves even though it's reserved for examples but can probably just use a particularly long root domain - we can probably handle updating the test if it ever gets actually registered.

Reply:
+1 this, we shouldn't hardcode ports in tests.

Reply:
Fixed.